core/arch/riscv/include/riscv.h

```diff
@@ -0,0 +1,283 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2022 NXP
+ */
+
+#ifndef RISCV_H
+#define RISCV_H
+
+#include <compiler.h>
+#include <encoding.h>
+#include <sys/cdefs.h>
+#include <stdint.h>
+#include <util.h>
+
+#define RISCV_XLEN_BITS		(__riscv_xlen)
+#define RISCV_XLEN_BYTES	(__riscv_xlen / 8)
+
+#define REGOFF(x)			((x) * RISCV_XLEN_BYTES)
+
+#if __riscv_xlen == 32
+#define STR       sw
+#define LDR       lw
+#else
+#define STR       sd
+#define LDR       ld
+#endif
+
+#ifndef __ASSEMBLER__
+
+#define get_field(reg, mask) (((reg) & \
```

Comment:
These macros seems quite useful. How about adding them to `lib/libutils/ext/include/util.h` as
GET_FIELD_U{32,64}() and SET_FIELD_U{32,64}()?

Reply:
Moved to a separate [PR#5444](https://github.com/OP-TEE/optee_os/pull/5444)